### PR TITLE
NEO-1171 AppLayout component Part2

### PR DESCRIPTION
**Pull request by enrique-prado**
**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work

Footer Area has been added, if provided it will render at the bottom of the screen.
The only area (prop) that is required is mainContent. Any optional areas that are not provided will not be rendered. See storybook stories in Layout category.

image

**IMPORTANT NOTES:**
- I'm getting unexpected errors in my basic render tests. It will take time to investigate a fix so I will add tests in Part 3.

### CODE REVIEW

File: src/components/Layout/AppLayout.tsx [hunk 1]
```diff
@@ -64,18 +60,19 @@ export const AppLayout = ({
       areas={mobileAreas || defaultMobileAreas}
       areasMd={desktopAreas || defaultDesktopAreas}
       areasLg={desktopAreas || defaultBigscreenAreas}
-      templateRows={"60px 1fr 30px"}
+      height={height}
+      templateRows={"auto 1fr auto"}
       templateCols={"1fr auto"}
-      templateColsMd={"250px 1fr auto"}
+      templateColsMd={"auto 1fr auto"}
```
Comment (mrazian85) [inline src/components/Layout/AppLayout.tsx:66]: Do we want/should we allow users to access & override these values?

**Comment by mrazian85** [comment]
I like this a lot! Left a question/discussion point

File: src/components/Layout/AppLayout.tsx (hunk 1 above)
Comment (enrique-prado) [reply src/components/Layout/AppLayout.tsx:66]: Good question. In a sense they could override the auto values by passing a leftPanel (or rightPanel) component with a fixed size... the "auto" will simply resize to the size of the component that was passed in.  The center column "1fr" will simply take whatever horizontal space is available.

Comment (enrique-prado) [reply src/components/Layout/AppLayout.tsx:66]: Hi Mo, I replied to your question in the PR. Let me know if that answers it or if you need more info. thanks.

Comment (mrazian85) [reply src/components/Layout/AppLayout.tsx:66]: Yes, thanks.

I am happy with this. I think that, other than needing some tests (which I assume will come later?), this Component in particular will require feedback from our users in order to better understand how they are using it, and for us to make the necessary improvements. So, the sooner we get this out in the wild the better.

**Review by mrazian85 — APPROVED** [review]

File: src/components/Layout/AppLayout.tsx (hunk 1 above)
Comment (enrique-prado) [reply src/components/Layout/AppLayout.tsx:66]: Yes, customer feedback will be key... and yes, the test will come later once I figure out a workaround for the failures.

File: src/components/Layout/AppLayout.tsx [hunk 2]
```diff
@@ -64,18 +60,19 @@ export const AppLayout = ({
       areas={mobileAreas || defaultMobileAreas}
       areasMd={desktopAreas || defaultDesktopAreas}
       areasLg={desktopAreas || defaultBigscreenAreas}
-      templateRows={"60px 1fr 30px"}
+      height={height}
+      templateRows={"auto 1fr auto"}
```
Comment (joe-s-avaya) [inline src/components/Layout/AppLayout.tsx:64]: nit: using `{"ish"}` is doing two calculations, strongly recommend updating all `{"` to just `"`
```suggestion
      templateRows="auto 1fr auto"
```

Comment (enrique-prado) [reply src/components/Layout/AppLayout.tsx]: Ah, good catch! done.

**Pull request merged**